src/GDCInternetAccount/model.tsx

```diff
@@ -0,0 +1,286 @@
+          resolve = () => {}
+          reject = () => {}
+          openLocationPromise = undefined
+        },
+        async openLocation(location: FileLocation) {
+          if (!openLocationPromise) {
+            // determine if the resource requires a token
+            const query = (location as AuthLocation).uri.split('/').pop()
+            const response = await fetch(
+              `${self.accountConfig.customEndpoint}/files/${query}?expand=index_files`,
+              {
+                method: 'GET',
+              },
+            )
+
+            if (!response.ok) {
+              const errorText = await response.text()
+              throw new Error(
+                `Network response failure: ${response.status} (${errorText})`,
+              )
+            }
+
+            const metadata = await response.json()
+            if (metadata) {
+              metadata.data.access === 'controlled'
+                ? self.setNeedsToken(true)
+                : self.setNeedsToken(false)
+            }
+
+            openLocationPromise = new Promise(async (r, x) => {
+              if (self.needsToken) {
+                const { session } = getRoot(self)
+                session.setDialogComponent(GDCTokenEntryForm, {
+                  internetAccountId: self.internetAccountId,
+                  handleClose: this.handleClose,
+                })
+              }
+              resolve = r
+              reject = x
+            })
+          }
+          return openLocationPromise
+        },
+        async handleRpcMethodCall(
+          location: FileLocation,
+          authenticationInfoMap: Record<string, string>,
+          args: {},
+        ) {
+          const token = authenticationInfoMap[self.internetAccountId]
+          if (!token) {
+            await this.openLocation(location)
+          }
+
+          try {
+            const query = (location as AuthLocation).uri.split('/').pop()
+            args = JSON.parse(JSON.stringify(args))
+            searchOrReplaceInArgs(
+              args,
+              'uri',
+              `${self.accountConfig.customEndpoint}/data/${query}`,
+            )
+          } catch (e) {
+            await this.handleError(authenticationInfoMap, e)
+          }
+
+          return args
+        },
+        async handleError(
+          authenticationInfoMap: Record<string, string>,
+          errorText: string,
+        ) {
+          const rootModel = getRoot(self)
+          rootModel.removeFromAuthenticationMap(
+            self.internetAccountId,
+            authenticationInfoMap,
+          )
+          return new Error(errorText)
+        },
+      }
+    })
+}
+
+const GDCTokenEntryForm = ({
+  internetAccountId,
+  handleClose,
+}: {
+  internetAccountId: string
+  handleClose: (arg?: string) => void
+}) => {
+  const [token, setToken] = useState('')
+  const classes = useStyles()
+
+  return (
+    <>
+      <Dialog open onClose={() => handleClose()} maxWidth="sm">
```

Comment:
instead of creating this component in here, just use the LoginDialog that already exists, import it and use that

(this is in here for ease of testing while the core auth is still in development)